zqd/handlers_test.go

```diff
@@ -287,6 +292,28 @@ func TestSpacePostNameOnly(t *testing.T) {
 	assert.Regexp(t, "^sp", sp.ID)
 }
 
+func TestSpacePostDuplicateName(t *testing.T) {
+	ctx := context.Background()
+	_, client, done := newCore(t)
+	defer done()
+	_, err := client.SpacePost(ctx, api.SpacePostRequest{Name: "test"})
+	require.NoError(t, err)
+	_, err = client.SpacePost(ctx, api.SpacePostRequest{Name: "test"})
+	require.Equal(t, api.ErrSpaceExists, err)
+}
+
+func TestSpacePutDuplicateName(t *testing.T) {
+	ctx := context.Background()
+	_, client, done := newCore(t)
+	defer done()
+	_, err := client.SpacePost(ctx, api.SpacePostRequest{Name: "test"})
+	require.NoError(t, err)
+	sp, err := client.SpacePost(ctx, api.SpacePostRequest{Name: "test1"})
+	require.NoError(t, err)
```

Comment:
Is there any need to create test1?

Reply:
Yep in order to test the space put endpoint I need 1) and existing space to collide with 2) another existing space to change the name to a colliding name.